test/uk/gov/hmrc/merchandiseinbaggagefrontend/controllers/EoriNumberControllerSpec.scala

```diff
@@ -0,0 +1,136 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.merchandiseinbaggagefrontend.controllers
+
+import play.api.test.Helpers._
+import uk.gov.hmrc.merchandiseinbaggagefrontend.model.core.{Eori, YesNo}
+import uk.gov.hmrc.merchandiseinbaggagefrontend.views.html.EoriNumberView
+
+import scala.concurrent.ExecutionContext.Implicits.global
+
+class EoriNumberControllerSpec extends DeclarationJourneyControllerSpec {
+
+  private lazy val controller =
+    new EoriNumberController(
+      controllerComponents, actionBuilder, declarationJourneyRepository, injector.instanceOf[EoriNumberView])
+
+  "onPageLoad" must {
+    val url = routes.EoriNumberController.onPageLoad().url
+    val getRequest = buildGet(url, sessionId)
+
+    behave like anEndpointRequiringASessionIdAndLinkedDeclarationJourneyToLoad(controller, url)
+
+    "redirect to /invalid-request" when {
+      "a declaration has been started but required answer has not been provided" in {
+        givenADeclarationJourneyIsPersisted(startedDeclarationJourney)
+
+        val result = controller.onPageLoad()(getRequest)
+
+        status(result) mustEqual SEE_OTHER
+        redirectLocation(result).get mustBe routes.InvalidRequestController.onPageLoad().url
+      }
+    }
+
+    "return OK and render the view" when {
+      "a declaration has been started and user is a trader" in {
+        givenADeclarationJourneyIsPersisted(startedDeclarationJourney.copy(maybeIsACustomsAgent = Some(YesNo.No)))
+
+        val result = controller.onPageLoad()(getRequest)
+        val content = contentAsString(result)
+
+        status(result) mustEqual OK
+        content must include("What is your EORI number?")
+      }
+
+      "a declaration has been started and user is an agent" in {
+        givenADeclarationJourneyIsPersisted(startedDeclarationJourney.copy(maybeIsACustomsAgent = Some(YesNo.Yes)))
+
+        val result = controller.onPageLoad()(getRequest)
+        val content = contentAsString(result)
+
+        status(result) mustEqual OK
+        content must include("What is the EORI number of the company importing the goods?")
+      }
+
+      "a declaration has been started and a value saved" in {
+        givenADeclarationJourneyIsPersisted(startedDeclarationJourney.copy(
+          maybeIsACustomsAgent = Some(YesNo.No),
+          maybeEori = Some(Eori("GB123467800000"))
+        ))
+
+        val result = controller.onPageLoad()(getRequest)
+        val content = contentAsString(result)
+
+        status(result) mustEqual OK
+        content must include("GB123467800000")
+      }
+    }
+  }
+
+  "onSubmit" must {
+    val url = routes.EoriNumberController.onSubmit().url
+    val postRequest = buildPost(url, sessionId)
+
+    behave like anEndpointRequiringASessionIdAndLinkedDeclarationJourneyToUpdate(controller, url)
+
+    "Redirect to /traveller-details" when {
```

Comment:
Not sure we need this test given the selenium spec. Does it cover anything different?

Reply:
Better safe than sorry. I feel like if I hadn't included both you would have asked for both 🙃

Reply:
Bad Paul!

I removed some controller tests the other day. I'd say only add what we need. 

Up to you.